pages/complete-login.js

```diff
@@ -29,17 +30,25 @@ const CompleteLoginPage = () => {
   async function attemptCompleteLogin () {
     setIsLoading(true)
     try {
-      const { e: encodedEmail, token } = router.query
-      if (!encodedEmail || !token) return
+      const { e: encodedEmail, token, code, state } = router.query
+      if ((!encodedEmail || !token) && (!code || !state)) return
       if (loginAttempted) {
         showError()
-        setIsLoading(false)
         return
       }
 
-      const email = decode(encodedEmail || '').toString()
       setLoginAttempted(true)
-      await auth.completeLogin({ email, token })
+
+      // If code and state are passed in, then it as a GH auth redirect
+      if (code && state) {
+        await auth.completeGHLogin({ code, state })
```

Comment:
yeah realizing now that i'm part way through the review -- the state logic needs to live server-side (i think)... since we are passing up code+state to API, the API should have been the one to give us the state var in the first place.

Reply:
moved to local storage